src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper_windows.cc

```diff
@@ -0,0 +1,32 @@
+/*
+ *
+ * Copyright 2016 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/iomgr/port.h"
+#if GRPC_ARES == 1 && defined(GPR_WINDOWS)
+
+#include "src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.h"
+
+bool grpc_ares_query_ipv6() {
+  /* TODO: do an "ipv6 available" check here?
```

Comment:
It's fairly straightforward to copy/paste the code from the posix version and adapt it for Windows. The idea is the same: create an IPv6 socket (using [socket()](https://docs.microsoft.com/en-us/windows/desktop/api/winsock2/nf-winsock2-socket), that returns a `SOCKET` instead of an int), check if it succeded, then try [bind()](https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-bind)ing it to `::1`, and finally [closesocket()](https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-closesocket) it to finalize the whole thing.

Reply:
done